In [1]:
import numpy as np
import math

In [2]:
disciplines = ['Проектування ІС', 'ЙМтаСОвІУС', 'КМ', 'ІПЗтаКР', 'АД', 'Англійська мова']

In [3]:
M = np.zeros((6, 6))
M[0, 1] = 1
M[0, 2] = 1
M[0, 3] = 1
M[0, 4] = 1
M[0, 5] = 1

M[1, 2] = 0.5
M[1, 3] = 0
M[1, 4] = 0.5
M[1, 5] = 0

M[2, 3] = 0
M[2, 4] = 0.5
M[2, 5] = 0

M[3, 4] = 1
M[3, 5] = 0.5

M[4, 5] = 0

for i in range(len(disciplines)):
    M[i, i] = 0.5
    for j in range(0, i):
        M[i, j] = 1 - M[j, i]

M

array([[0.5, 1. , 1. , 1. , 1. , 1. ],
       [0. , 0.5, 0.5, 0. , 0.5, 0. ],
       [0. , 0.5, 0.5, 0. , 0.5, 0. ],
       [0. , 1. , 1. , 0.5, 1. , 0.5],
       [0. , 0.5, 0.5, 0. , 0.5, 0. ],
       [0. , 1. , 1. , 0.5, 1. , 0.5]])

In [4]:
score = {}

for discipline in disciplines:
    i = disciplines.index(discipline)
    s = np.sum(M[i])
    score[discipline] = s
    print(f'Сума для {discipline}: {s}')

Сума для Проектування ІС: 5.5
Сума для ЙМтаСОвІУС: 1.5
Сума для КМ: 1.5
Сума для ІПЗтаКР: 4.0
Сума для АД: 1.5
Сума для Англійська мова: 4.0


In [5]:
pairs = math.comb(len(disciplines), 2)
print(f'Кількість пар: {pairs}')

Кількість пар: 15


In [6]:
score_list = list(score.values())
score_sum = np.sum(score_list)
print(f'Сума всіх оцінок: {score_sum}')

Сума всіх оцінок: 18.0


In [7]:
score_mean = np.mean(score_list)
print(f'Середня оцінка: {score_mean}')

Середня оцінка: 3.0


In [8]:
S = np.sum((score_list - score_mean)**2)
n = len(score_list)

W = (12 * S)/(n**2 * (n**2 - 1))
print(f'W= {W}')

W= 0.14285714285714285


Оскільки коефіцієнт Кендела узгодженості Кендела ~ 0.14, то це свідчить про те, що оцінки не узгоджені та судження є суперечливими.